### PR TITLE
add package Luabridge3

### PR DESCRIPTION
**Pull request by tokomine**
Referenced the implementation of vcpkg:
https://github.com/microsoft/vcpkg/blob/master/ports/luabridge3/portfile.cmake

```
# Copy the header files
file(COPY "${SOURCE_PATH}/Source/LuaBridge" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}")
```

the header file path is
```
#include "luabridge3/LuaBridge/LuaBridge.h"
```

### CODE REVIEW

File: packages/l/luabridge3/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        os.cp("Source", package:installdir() .. "/include/luabridge3")
+    end)
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+            #define LUA_VERSION_NUM
+            #include <luabridge3/LuaBridge/detail/Errors.h>
+            void test()
+            {
+                typedef luabridge::ErrorCode __type_luabridge__ErrorCode;
```
Comment (waruqi) [inline packages/l/luabridge3/xmake.lua:22]: please add some api call to test link.

File: packages/l/luabridge3/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package("luabridge3")
+
```
Comment (waruqi) [inline packages/l/luabridge3/xmake.lua:2]: if it's header only, please set `set_kind("library", {headeronly = true})`

File: packages/l/luabridge3/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    set_homepage("https://kunitoki.github.io/LuaBridge3")
+    set_description("LuaBridge3 is a lightweight and dependency-free library for mapping data, functions, and classes back and forth between C++ and Lua.")
+    set_license("MIT Licensed")
```
Comment (waruqi) [inline packages/l/luabridge3/xmake.lua:5]: `set_license("MIT")

File: packages/l/luabridge3/xmake.lua (hunk 1 above)
Comment (tokomine) [reply packages/l/luabridge3/xmake.lua:23]: 这个库需要依赖lua，不能单独include，这种情况怎么写test？例如：
```C++
#pragma once

#include "Lua/LuaLibrary.h"

#include "LuaBridge/LuaBridge.h"

#include <gtest/gtest.h>

#if LUABRIDGE_ON_LUAU
#include "../../ThirdParty/luau/Compiler/include/luacode.h"
#include "../../ThirdParty/luau/Common/include/Luau/Common.h"
```

**Pull request merged**